luigi/parameter.py

```diff
@@ -1115,10 +1137,14 @@ def normalize(self, value):
         """
         Ensure that dictionary parameter is converted to a FrozenOrderedDict so it can be hashed.
         """
-        frozen_value = recursively_freeze(value)
         if self.schema is not None:
-            jsonschema.validate(instance=recursively_unfreeze(frozen_value), schema=self.schema)
-        return frozen_value
+            unfrozen_value = recursively_unfreeze(value)
+            try:
+                self.schema.validate(unfrozen_value)
+                value = unfrozen_value  # Validators may update the instance inplace
```

Comment:
I defined the `unfrozen_value` variable for clarity but we could just overwrite the `value` variable at line `1141`. What do you prefer?